### PR TITLE
Working on TLD 2.06

### PR DESCRIPTION
**Pull request by dommrogers**

### CODE REVIEW

File: LocalizationUtilities/LocalizationPatch.cs [hunk 1]
```diff
@@ -6,14 +6,17 @@ namespace LocalizationUtilities;
...
+		private static void Postfix()
 		{
-			AddOrUpdate(strTable, set);
+			StringTable strTable = Localization.s_CurrentLanguageStringTable;
+			foreach (LocalizationSet set in LocalizationManager.Localizations)
+			{
+				AddOrUpdate(strTable, set);
+			}
 		}
 	}
```
Comment (ds5678) [inline LocalizationUtilities/LocalizationPatch.cs]: Revert all this back to what I had and add a `[HarmonyPatch]` attribute to `LocalizationPatch`. When I was porting this mod, I forgot to add the attribute.

File: LocalizationUtilities/LocalizationUtilities.csproj [hunk 2]
```diff
@@ -22,36 +17,37 @@
...
+	
 	<PropertyGroup>
-		<MelonLoaderPath>$(TheLongDarkPath)/MelonLoader/net6</MelonLoaderPath>
-		<ManagedPath>$(TheLongDarkPath)/MelonLoader/Managed</ManagedPath>
-		<Il2CppPath>$(TheLongDarkPath)/MelonLoader/Il2CppAssemblies</Il2CppPath>
-		<ModsPath>$(TheLongDarkPath)/Mods</ModsPath>
-		<AssemblySearchPaths>$(AssemblySearchPaths);$(MelonLoaderPath);$(ManagedPath);$(Il2CppPath);$(ModsPath);</AssemblySearchPaths>
+		<Authors>ds5678</Authors>
+		<Version>1.1.0.0</Version>
+		<AssemblyVersion>1.1.0.0</AssemblyVersion>
```
Comment (ds5678) [inline LocalizationUtilities/LocalizationUtilities.csproj]: 2.0.0

File: LocalizationUtilities/LocalizationManager.cs [hunk 3]
```diff
@@ -72,16 +37,23 @@ private static string GetText(TextAsset textAsset)
...
 		}
-		return sb.ToString();
+		string str = sb.ToString();
+
+		int startFrom = str.IndexOf(@"{");
+		if (startFrom > 0)
+		{
+			str = str.Substring(startFrom);
+		}
+		return str;
```
Comment (ds5678) [inline LocalizationUtilities/LocalizationManager.cs]: When I initially made the mod, I didn't know what I was doing, and newer API's were not available. There is a substantially better way to do this.

```cs
const byte leftCurlyBracket = (byte)'{';
byte[] bytes = textAsset.bytes;
int index = Array.IndexOf(bytes, leftCurlyBracket);
if (index < 0)
{
    throw new ArgumentException("TextAsset has no Json content.", nameof(textAsset));
}
return Encoding.UTF8.GetString(new ReadOnlySpan<byte>(bytes, index, bytes.Length - index);
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      with:
+        name: LocalizationUtilities
+        path: ./LocalizationUtilities/bin/Release/net6.0/LocalizationUtilities.dll
+        if-no-files-found: error
+        
+    - name: Upload
+      uses: actions/upload-artifact@v3
+      with:
+        name: LocalizationUtilities_nupkg
+        path: ./LocalizationUtilities/bin/Release/ds5678.Modding.TLD.LocalizationUtilities.1.1.0.nupkg
```
Comment (ds5678) [inline .github/workflows/build.yml]: I think `./LocalizationUtilities/bin/Release/*.nupkg` or `./LocalizationUtilities/bin/Release/**.nupkg` might work here.

If not, change to 2.0.0 similar to the other places.

File: LocalizationUtilities/BuildInfo.cs [hunk 5]
```diff
@@ -6,6 +6,6 @@ public static class BuildInfo
 	public const string Description = "A utility mod for localized text."; // Description for the Mod.  (Set as null if none)
 	public const string Author = "ds5678"; // Author of the Mod.  (MUST BE SET)
 	public const string Company = null; // Company that made the Mod.  (Set as null if none)
-	public const string Version = "1.0.0"; // Version of the Mod.  (MUST BE SET)
+	public const string Version = "1.1.0"; // Version of the Mod.  (MUST BE SET)
```
Comment (ds5678) [inline LocalizationUtilities/BuildInfo.cs]: 2.0.0

**Review by ds5678 — CHANGES_REQUESTED** [review]
Looking good. Just a few small tweaks.

File: LocalizationUtilities/LocalizationPatch.cs (hunk 1 above)
Comment (dommrogers) [reply LocalizationUtilities/LocalizationPatch.cs]: Done

File: LocalizationUtilities/LocalizationManager.cs (hunk 3 above)
Comment (dommrogers) [reply LocalizationUtilities/LocalizationManager.cs]: Done & tested.

File: LocalizationUtilities/LocalizationUtilities.csproj (hunk 2 above)
Comment (dommrogers) [reply LocalizationUtilities/LocalizationUtilities.csproj]: Done

**Review by ds5678 — APPROVED** [review]

**Pull request merged**